### PR TITLE
Refactor/folder structure

### PR DESCRIPTION
**Pull request by jalarg**
# Summary
- Add pages (including _app.tsx,  _document.tsx and and index.tsx files) and styles folder
- Refactor types, components, commons and assets folder

# Details
- The app now uses the _pages folder_ with the _app.tsx, _document.tsx and index.tsx.

# Evidence 

## Folders order: 

Folder order

### CODE REVIEW

File: cypress/e2e/home.cy.ts [hunk 1]
```diff
@@ -1,8 +1,67 @@
...
-describe("Navigation", () => {
-  it("Go to the website:", () => {
-    console.log(projectUrl, "ProjectURL");
-    cy.visit(projectUrl);
+beforeEach(() => {
+  cy.visit(projectUrl);
+});
+
+describe("Home: UI and functionality:", () => {
+  it("Home UI: Stellar logo", () => {
```
Comment (Genebson) [inline cypress/e2e/home.cy.ts]: We write test messages like this: `"Should the Stellar logo be visible"`

File: cypress/e2e/home.cy.ts [hunk 2]
```diff
@@ -1,8 +1,67 @@
...
+    cy.get('[data-cy="stellar-home-logo"]').should("be.visible");
+  });
```
Comment (Genebson) [inline cypress/e2e/home.cy.ts:18]: Add an enter here, please.

File: cypress/e2e/home.cy.ts [hunk 3]
```diff
@@ -1,8 +1,67 @@
...
+    cy.get('[data-cy="rocket-description-1"]')
+      .should("be.visible")
+      .and("contain", "The easiest way to connect with a wallet");
+    cy.get('[data-cy="rocket-description-2"]')
+      .should("be.visible")
+      .and("contain", "Sign in methods");
+  });
+
+  it("Button connect: Should show a button to generate connect", () => {
+    cy.get('[data-cy="connect"]')
```
Comment (Genebson) [inline cypress/e2e/home.cy.ts]: This could be an alias.

File: cypress/e2e/home.cy.ts [hunk 4]
```diff
@@ -1,8 +1,67 @@
...
+      .should("be.visible")
+      .and("contain", "Connect with a secret key");
+    cy.get('[data-cy="connect"]').should("exist");
+  });
+
+  it("Button create wallet: Should show a button to generate a pair of new keys", () => {
+    cy.get('[data-cy="register"]').should("exist");
```
Comment (Genebson) [inline cypress/e2e/home.cy.ts]: Same here, this could be an alias, and the ones below that we repeat.

File: cypress/e2e/home.cy.ts [hunk 5]
```diff
@@ -1,8 +1,67 @@
...
+    cy.get('[data-cy="public-key"]').should("exist");
+    cy.get('[data-cy="private-key"]').should("exist");
+    cy.get('[data-cy="confirm-wallet"]').should("exist");
+    cy.get('[data-cy="go-back"]').should("exist");
+    cy.get('[data-cy="confirm-wallet"]').should("have.text", "Confirm wallet");
+    cy.get('[data-cy="go-back"]').should("have.text", "Go back");
+  });
+});
+
+describe(" Navbar: UI and functionality:", () => {
```
Comment (Genebson) [inline cypress/e2e/home.cy.ts:57]: There's a blank space here, delete it, please.

File: cypress/e2e/home.cy.ts [hunk 6]
```diff
@@ -1,8 +1,67 @@
...
+  it("Check Stellar Icon functionality", () => {
```
Comment (Genebson) [inline cypress/e2e/home.cy.ts]: `Should...`

File: cypress/e2e/home.cy.ts [hunk 7]
```diff
@@ -1,8 +1,67 @@
...
+    // cy.visit(projectUrl);
```
Comment (Genebson) [inline cypress/e2e/home.cy.ts]: Delete this, please.

File: cypress/e2e/home.cy.ts [hunk 8]
```diff
@@ -1,8 +1,67 @@
...
+    cy.get('[data-cy="navbar-container"]').should("exist").and("be.visible");
+    cy.get('[data-cy="stellar-icon"]')
+      .should("be.visible")
+      .and("have.attr", "href", stellarIconUrl);
+  });
+
+  it("Check theme functionality", () => {
```
Comment (Genebson) [inline cypress/e2e/home.cy.ts]: `Should...`

File: cypress/e2e/home.cy.ts [hunk 9]
```diff
@@ -1,8 +1,67 @@
...
+    // cy.visit(projectUrl);
```
Comment (Genebson) [inline cypress/e2e/home.cy.ts]: Delete this, please.

File: cypress/support/constants.ts [hunk 10]
```diff
@@ -0,0 +1,2 @@
+export const projectUrl = Cypress.env("BASE_URL");
+export const stellarIconUrl = "https://stellar.org/";
```
Comment (Genebson) [inline cypress/support/constants.ts]: This is not an icon but a URL
```suggestion
export const stellarUrl = "https://stellar.org/";
```

File: src/components/Navbar.tsx [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+import { useEffect } from "react";
+
+const Navbar = () => {
+  const [theme, setTheme] = useState<string>("light");
+
+  const handleChangeTheme = () =>
+    setTheme((prevTheme) => (prevTheme === "light" ? "dark" : "light"));
+
+  useEffect(() => {
+    console.log(theme);
```
Comment (Genebson) [inline src/components/Navbar.tsx]: Remove this, please.

File: src/components/Register.tsx [hunk 12]
```diff
@@ -1,11 +1,12 @@
-import Button from "../commons/Button";
+import Button from "../components/commons/Button";
 import { ISetKeyPair } from "../types/types";
 import registerHandler from "../actions/registerHandler";
 
 const Register = ({ setSecret, setPublicKey }: ISetKeyPair) => {
   return (
     <div className="flex justify-center m-5">
       <Button
+        cypressId="register"
```
Comment (Genebson) [inline src/components/Register.tsx]: What's this?

File: src/types/types.d.ts [hunk 13]
```diff
@@ -6,6 +6,7 @@ export interface InputProps {
   placeholder?: string;
   required?: boolean;
   readOnly?: boolean;
+  cypressId?: string;
```
Comment (Genebson) [inline src/types/types.d.ts]: I think this is not necessary.

File: cypress/e2e/home.cy.ts (hunk 1 above)
Comment (jalarg) [reply cypress/e2e/home.cy.ts]: Issue solved in commit f0a74323da3f19ea9b248e1159579dd2625abb9d

File: cypress/e2e/home.cy.ts (hunk 2 above)
Comment (jalarg) [reply cypress/e2e/home.cy.ts:18]: issue solved in commit f0a74323da3f19ea9b248e1159579dd2625abb9d.

File: cypress/e2e/home.cy.ts (hunk 3 above)
Comment (jalarg) [reply cypress/e2e/home.cy.ts]: Started using alias, check commit f0a74323da3f19ea9b248e1159579dd2625abb9d.

File: cypress/e2e/home.cy.ts (hunk 4 above)
Comment (jalarg) [reply cypress/e2e/home.cy.ts]: idem prior comment, see commit f0a74323da3f19ea9b248e1159579dd2625abb9d.

File: cypress/e2e/home.cy.ts (hunk 5 above)
Comment (jalarg) [reply cypress/e2e/home.cy.ts:57]: solved.

File: cypress/e2e/home.cy.ts (hunk 7 above)
Comment (jalarg) [reply cypress/e2e/home.cy.ts]: Solved.

File: cypress/e2e/home.cy.ts (hunk 6 above)
Comment (jalarg) [reply cypress/e2e/home.cy.ts]: Solved in commit f0a74323da3f19ea9b248e1159579dd2625abb9d.

File: cypress/e2e/home.cy.ts (hunk 8 above)
Comment (jalarg) [reply cypress/e2e/home.cy.ts]: Solved in commit f0a74323da3f19ea9b248e1159579dd2625abb9d.

File: cypress/e2e/home.cy.ts (hunk 9 above)
Comment (jalarg) [reply cypress/e2e/home.cy.ts]: solved in commit f0a74323da3f19ea9b248e1159579dd2625abb9d.

File: src/components/Navbar.tsx (hunk 11 above)
Comment (jalarg) [reply src/components/Navbar.tsx]: Solved in commit 0109e918ef7b246d5657c1ca036c4711e157b3c8.

File: cypress/support/constants.ts (hunk 10 above)
Comment (jalarg) [reply cypress/support/constants.ts]: Solved in commit 7efb5acd746344ce0dd24e6411532b17f3975581.

**Review by Genebson — APPROVED** [review]

**Pull request merged**